File: 프로그래머스/2/64065. 튜플/튜플.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <set>

using namespace std;

vector<int> solution(string s) {
    vector<vector<int>> check;

    vector<int> temp;
    string num;

    for (char c : s) {
        if (isdigit(c)) {
            num += c;
        }
        else {
            if (!num.empty()) {
                temp.push_back(stoi(num));
                num.clear();
            }

            if (c == '}') {
                if (!temp.empty()) {
                    check.push_back(temp);
                    temp.clear();
                }
            }
        }
    }

    sort(check.begin(), check.end(),
         [](vector<int>& a, vector<int>& b) {
             return a.size() < b.size();
         });

    vector<int> answer;
    set<int> used;

    for (auto& i : check) {
        for (int n : i) {
            if (used.find(n) == used.end()) {
                answer.push_back(n);
                used.insert(n);
            }
        }
    }

    return answer;
}

File: 프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    s = s[2:-2]
    check = s.split("},{")
    check.sort(key=len)
    check = [list(map(int,i.split(","))) for i in check]
    
    answer = []
    used = set()
    
    for i in check:
        for n in i:
            if n not in used:
                answer.append(n)
                used.add(n)
    
    return answer